File: programs/pyeos/tests/lua/hello/hello.lua
function apply(receiver, account, act)
    code = receiver
    scope = receiver
    table = receiver
    payer = receiver
    id = receiver
--    print(id)

    if act == N('sayhello') then
        msg = read_action_data()
--        print(msg)
        itr = db_find_i64(code, scope, table, id)
--        print("itr:", itr)
      
        if itr >= 0 then
            data = db_get_i64(itr)
--            print(data)
        else
            db_store_i64(scope, table, payer, id, msg)
        end

--        while true do
--        end
    end
    

     return 1
end

File: programs/pyeos/tests/java/hello/Hello.java
import java.io.*;

public class Hello extends Contract {
	public Hello() {
		
	}
	public int apply(long receiver, long account, long act) {
		long code = receiver;
		long scope = receiver;
		long payer = code;
		long table_id = code;
		long key = code;
		if (act != VMJava.s2n("sayhello")) {
			return 0;
		}
/*
		try {
			System.getProperty("user.dir");

			PrintWriter out = new PrintWriter(new FileWriter("OutFile.txt"));
			out.println("hello,world");
			out.close();
		} catch (java.io.UnsupportedEncodingException ex) {
			System.out.println(ex);
		} catch (IOException ex) {
			
		}
*/

		byte[] data;
		int itr = VMJava.db_find_i64(code, scope, table_id, key);
		data = VMJava.read_action_data();
//			System.out.println("+++++data.length:"+data.length);
//			System.out.println("++++"+new String(data, "UTF-8"));

		if (itr < 0) {
			VMJava.db_store_i64(scope, payer, table_id, key, data);
		} else {
			byte[] old_data = VMJava.db_get_i64(itr);
//				System.out.println("+++++old_data.length:"+old_data.length);
			VMJava.db_update_i64(itr, payer, data);
		}

//		System.out.println("hello, java world!");
		return 1;
	}
}
